pkg/compose/logs.go

```diff
@@ -106,6 +114,10 @@ func (s *composeService) Logs(
 		eg.Go(func() error {
 			return monitor.Start(ctx)
 		})
```

Comment:
```suggestion
		eg.Go(func() error {
			return monitor.Start(ctx)
		})
```

Sorry if I jump in, but I got bitten by the recent change done by #13062.
With the fix you propose in this PR I'm afraid we'll have a goroutine leak. I think the monitor.Start will keep running.

Considering only this codebase's use case for the Logs function, the leak won't be a big deal because you always quit the process. However, this will become a problem for other users who use this function and do not want to terminate the process.

I think it would be great if we could pass to the `monitor.Start`/`newMonitor` something to make it go back the old behaviour (quitting the monitor upon context cancellation)